ocis-pkg/handlers/checker.go

```diff
@@ -43,19 +45,21 @@ func (c CheckHandlerConfiguration) WithLogger(l log.Logger) CheckHandlerConfigur
 }
 
 // WithCheck sets a check for the CheckHandlerConfiguration.
-func (c CheckHandlerConfiguration) WithCheck(name string, f check) CheckHandlerConfiguration {
-	if _, ok := c.Checks[name]; ok {
+func (c CheckHandlerConfiguration) WithCheck(name string, check checker) CheckHandlerConfiguration {
+	if _, ok := c.checks[name]; ok {
 		c.logger.Panic().Str("check", name).Msg("check already exists")
 	}
 
-	c.Checks[name] = f
+	c.checks = maps.Clone(c.checks) // prevent propagated check duplication, maps are references;
+	c.checks[name] = check
+
 	return c
 }
 
-// WithInheritedChecksFrom appends the checks from another CheckHandlerConfiguration.
-func (c CheckHandlerConfiguration) WithInheritedChecksFrom(other CheckHandlerConfiguration) CheckHandlerConfiguration {
-	for name, check := range other.Checks {
-		c.WithCheck(name, check)
+// WithChecks adds multiple checks to the CheckHandlerConfiguration.
+func (c CheckHandlerConfiguration) WithChecks(checks checkers) CheckHandlerConfiguration {
```

Comment:
This is easily confused with `WithChecks` at least rename it to `WithChecksFrom`

Reply:
I’m not a big fan at all the check handler can inherit from other handlers, I still think all those things should happen in the configuration only.

Is we name it ‘from’ it indicates that it uses the checks from xyz and not the checks directly, but if you want it that way, can you please push it? I’m not at my desk at the moment